src/libraries/System.Memory/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -588,52 +588,71 @@ private static unsafe void Ssse3Decode(ref byte* srcBytes, ref byte* destBytes,
             do
             {
                 AssertRead<Vector128<sbyte>>(src, srcStart, sourceLength);
-                Vector128<sbyte> str = Sse2.LoadVector128(src).AsSByte();
+                Vector128<byte> str = Vector128.LoadUnsafe(ref *src);
 
                 // lookup
-                Vector128<sbyte> hiNibbles = Sse2.And(Sse2.ShiftRightLogical(str.AsInt32(), 4).AsSByte(), mask2F);
-                Vector128<sbyte> loNibbles = Sse2.And(str, mask2F);
-                Vector128<sbyte> hi = Ssse3.Shuffle(lutHi, hiNibbles);
-                Vector128<sbyte> lo = Ssse3.Shuffle(lutLo, loNibbles);
+                Vector128<byte> hiNibbles = Vector128.ShiftRightLogical(str.AsInt32(), 4).AsByte() & mask2F;
+                Vector128<byte> hi = SimdShuffle(lutHi, hiNibbles, mask8F);
+                Vector128<byte> lo = SimdShuffle(lutLo, str, mask8F);
 
                 // Check for invalid input: if any "and" values from lo and hi are not zero,
                 // fall back on bytewise code to do error checking and reporting:
-                if (Sse2.MoveMask(Sse2.CompareGreaterThan(Sse2.And(lo, hi), zero)) != 0)
+                if ((lo & hi) != Vector128<byte>.Zero)
                     break;
 
-                Vector128<sbyte> eq2F = Sse2.CompareEqual(str, mask2F);
-                Vector128<sbyte> shift = Ssse3.Shuffle(lutShift, Sse2.Add(eq2F, hiNibbles));
+                Vector128<byte> eq2F = Vector128.Equals(str, mask2F);
```

Comment:
and here too: `str == mask2F`.

Reply:
This one can't be unless its immediately followed by the relevant `ExtractMostSignificantBits` and corresponding compare sequence.

`==` is equivalent to `EqualsAll` and returns a `bool` not a `vector`